noir-projects/noir-contracts/contracts/token_bridge_contract/src/main.nr

```diff
@@ -18,38 +22,34 @@ pub contract TokenBridge {
 
     use dep::token::Token;
 
-    use dep::aztec::macros::{
-        functions::{initializer, internal, private, public, view},
-        storage::storage,
-    };
+    use dep::aztec::macros::{functions::{initializer, private, public, view}, storage::storage};
     // docs:end:token_bridge_imports
 
-    // TODO: This can be optimized by storing the values in Config struct in 1 PublicImmutable (less merkle proofs).
     // docs:start:token_bridge_storage_and_constructor
     // Storage structure, containing all storage, and specifying what slots they use.
     #[storage]
     struct Storage<Context> {
-        token: PublicImmutable<AztecAddress, Context>,
-        portal_address: PublicImmutable<EthAddress, Context>,
+        config: PublicImmutable<Config, Context>,
     }
 
     // Constructs the contract.
     #[public]
     #[initializer]
-    fn constructor(token: AztecAddress, portal_address: EthAddress) {
-        storage.token.initialize(token);
-        storage.portal_address.initialize(portal_address);
+    fn constructor(token: AztecAddress, portal: EthAddress) {
+        storage.config.initialize(Config { token, portal });
     }
     // docs:end:token_bridge_storage_and_constructor
 
     #[private]
-    fn get_portal_address() -> EthAddress {
-        storage.portal_address.read()
+    #[view]
+    fn get_config() -> Config {
```

Comment:
Replaced `get_portal_address` and `get_token` funcs to a single `get_config` as it it declutters this contract and it also results in optimizations in uniswap contract because there we need both values in the single function.